src/renderer/views/Subscriptions/Subscriptions.js

```diff
@@ -183,86 +195,66 @@ export default defineComponent({
           }
         }
 
-        videoList = videoList.concat(videos)
         channelCount++
-        const percentageComplete = (channelCount / this.activeSubscriptionList.length) * 100
+        const percentageComplete = (channelCount / channelsToLoadFromRemote.length) * 100
         this.setProgressBarPercentage(percentageComplete)
+        this.updateSubscriptionsCacheForOneChannel({
+          channelId: channel.id,
+          videos: videos,
+        })
+        return videos
+      }))).flatMap((o) => o)
+      videoList.push(...videoListFromRemote)
 
-        if (channelCount === this.activeSubscriptionList.length) {
-          videoList = await Promise.all(videoList.sort((a, b) => {
-            return b.publishedDate - a.publishedDate
-          }))
-          if (this.hideLiveStreams) {
-            videoList = videoList.filter(item => {
-              return (!item.liveNow && !item.isUpcoming)
-            })
-          }
-          if (this.hideUpcomingPremieres) {
-            videoList = videoList.filter(item => {
-              if (item.isRSS) {
-                // viewCount is our only method of detecting premieres in RSS
-                // data without sending an additional request.
-                // If we ever get a better flag, use it here instead.
-                return item.viewCount !== '0'
-              }
-              // Observed for premieres in Local API Subscriptions.
-              return item.durationText !== 'PREMIERE'
-            })
-          }
-          const profileSubscriptions = {
-            activeProfile: this.activeProfile._id,
-            videoList: videoList,
-            errorChannels: this.errorChannels
-          }
+      this.updateVideoListAfterProcessing(videoList)
+      this.isLoading = false
+      this.updateShowProgressBar(false)
+    },
 
-          this.videoList = await Promise.all(videoList.filter((video) => {
-            if (this.hideWatchedSubs) {
-              const historyIndex = this.historyCache.findIndex((x) => {
-                return x.videoId === video.videoId
-              })
+    updateVideoListAfterProcessing(videoList) {
+      // Filtering and sorting based in preference
+      videoList.sort((a, b) => {
+        return b.publishedDate - a.publishedDate
+      })
+      if (this.hideLiveStreams) {
+        videoList = videoList.filter(item => {
+          return (!item.liveNow && !item.isUpcoming)
+        })
+      }
+      if (this.hideUpcomingPremieres) {
+        videoList = videoList.filter(item => {
+          if (item.isRSS) {
+            // viewCount is our only method of detecting premieres in RSS
+            // data without sending an additional request.
+            // If we ever get a better flag, use it here instead.
+            return item.viewCount !== '0'
+          }
+          // Observed for premieres in Local API Subscriptions.
+          return item.durationText !== 'PREMIERE'
```

Comment:
Invidious has:
  "isUpcoming": Bool,
or
  "premiereTimestamp": Int64?,

Reply:
I am waiting for https://github.com/FreeTubeApp/FreeTube/pull/3673 to be merged before I push my local updated code :P

Reply:
Wait, shouldn't this be in https://github.com/FreeTubeApp/FreeTube/pull/3673 then???

Reply:
Follow up pr will be created (as this is a different bug that what 3673 is fixing)